conda/core/python_dist.py

```diff
@@ -0,0 +1,1026 @@
+                    if isfile(fpath):
+                        metadata_path = fpath
+                        break
+            elif isfile(path):
+                # '<pkg>.egg-info' file contains metadata directly
+                filenames = ['.egg-info']
+                if metadata_filenames:
+                    filenames.extend(metadata_filenames)
+                assert any(path.endswith(filename) for filename in filenames)
+                metadata_path = path
+            else:
+                # `path` does not exist
+                warnings.warn("Metadata path not found", MetadataWarning)
+        else:
+            warnings.warn("Metadata path not found", MetadataWarning)
+
+        return metadata_path
+
+    @classmethod
+    def _message_to_dict(cls, message):
+        """
+        Convert the RFC-822 headers data into a dictionary.
+
+        `message` is an email.parser.Message instance.
+
+        The canonical method to transform metadata fields into such a data
+        structure is as follows:
+          - The original key-value format should be read with
+            email.parser.HeaderParser
+          - All transformed keys should be reduced to lower case. Hyphens
+            should be replaced with underscores, but otherwise should retain
+            all other characters
+          - The transformed value for any field marked with "(Multiple-use")
+            should be a single list containing all the original values for the
+            given key
+          - The Keywords field should be converted to a list by splitting the
+            original value on whitespace characters
+          - The message body, if present, should be set to the value of the
+            description key.
+          - The result should be stored as a string-keyed dictionary.
+        """
+        new_data = odict()
+
+        if message:
+            for key, value in message.items():
+
+                if key in cls.MULTIPLE_USE_KEYS:
+                    new_key = cls.MULTIPLE_USE_KEYS[key]
+                    if new_key not in new_data:
+                        new_data[new_key] = [value]
+                    else:
+                        new_data[new_key].append(value)
+
+                elif key in cls.SINGLE_USE_KEYS:
+                    new_key = cls.SINGLE_USE_KEYS[key]
+                    new_data[new_key] = value
+
+            # TODO: Handle license later on for convenience
+
+        return new_data
+
+    @classmethod
+    def _read_metadata(cls, fpath):
+        """
+        Read the original format which is stored as RFC-822 headers.
+        """
+        data = odict()
+        if fpath and isfile(fpath):
+            parser = email.parser.HeaderParser()
+
+            # FIXME: Is this a correct assumption for the encoding?
+            # This was needed due to some errors on windows
+            with io.open(fpath, 'r', encoding='utf-8') as fp:
+                data = parser.parse(fp)
```

Comment:
@kalefranz thoughts?

Reply:
Use `from ..common.compat import open`. And then no `encoding=`.

Reply:
Will do